src/Development/IDE/GHC/CPP.hs

```diff
@@ -139,7 +141,11 @@ getBackendDefs :: DynFlags -> IO [String]
 getBackendDefs dflags | hscTarget dflags == HscLlvm = do
     llvmVer <- figureLlvmVersion dflags
     return $ case llvmVer of
-#if MIN_GHC_API_VERSION(8,8,0)
+#if MIN_GHC_API_VERSION(8,8,2)
+               Just v
+                 | m:n:_ <- llvmVersionList v -> [ "-D__GLASGOW_HASKELL_LLVM__=" ++ format (m, n) ]
+                 | n:_   <- llvmVersionList v -> [ "-D__GLASGOW_HASKELL_LLVM__=" ++ format (n, 0) ]
+#elif MIN_GHC_API_VERSION(8,8,0)
```

Comment:
This matches the way this is written in GHC and is a bit easier to read imho: https://gitlab.haskell.org/ghc/ghc/blob/ghc-8.8/compiler/main/DriverPipeline.hs#L2174

Reply:
Aha, good call 👍

Reply:
Shame they didn’t export that.

Reply:
The whole doCpp is exported from GHC 8.10 and above, so the whole thing can disappear.

Reply:
@ndmitchell Do you mean that we should do that in this PR? Cos I’m not really set up to test against 8.10.* just now.

Reply:
No - we should leave it as it is for now. Once 8.10 is actually out we should do some conditional stuff - more just pointing out there is a long term plan for getting rid of it all.

Reply:
No, we need to support older versions so this can’t go away completely in particular not the code for `8.8.2`. We can however, use the definition from GHC in `8.10` but let’s please keep `8.10` support for a separate PR. I expect that there are a bunch of other issues anyway.